src/cascade/model/priors.py

```diff
@@ -0,0 +1,178 @@
+            dict(name=other.name, **other.parameters()).items()
+        )
+
+    def __repr__(self):
+        return f"<{type(self).__name__} {self.parameters()}>"
+
+
+def _validate_bounds(lower, mean, upper):
+    if not lower <= mean <= upper:
+        raise ValueError("Bounds are inconsistent")
+
+
+def _validate_standard_deviation(standard_deviation):
+    if standard_deviation < 0:
+        raise ValueError("Standard deviation must be positive")
+
+
+def _validate_nu(nu):
+    if nu < 0:
+        raise ValueError("Nu must be positive")
+
+
+class UniformPrior(_Prior):
+    density = "uniform"
+
+    def __init__(self, lower, upper, mean=None, name=None):
+        super().__init__(name=name)
+        if mean is None:
+            mean = (upper - lower) / 2 + lower
+        _validate_bounds(lower, mean, upper)
+
+        self.lower = lower
+        self.upper = upper
+        self.mean = mean
+
+    def _parameters(self):
+        return {"lower": self.lower, "upper": self.upper, "mean": self.mean}
+
+
+class ConstantPrior(_Prior):
+    density = "uniform"
+
+    def __init__(self, value, name=None):
+        super().__init__(name=name)
+        self.value = value
+
+    def _parameters(self):
+        return {"lower": self.value, "upper": self.value, "mean": self.value}
+
+
+class GaussianPrior(_Prior):
+    density = "gaussian"
+
+    def __init__(self, mean, standard_deviation, lower=float("-inf"), upper=float("inf"), name=None):
+        super().__init__(name=name)
+        _validate_bounds(lower, mean, upper)
+        _validate_standard_deviation(standard_deviation)
+
+        self.lower = lower
+        self.upper = upper
+        self.mean = mean
+        self.standard_deviation = standard_deviation
+
+    def _parameters(self):
+        return {"lower": self.lower, "upper": self.upper, "mean": self.mean, "std": self.standard_deviation}
+
+
+class LaplacePrior(GaussianPrior):
+    density = "laplace"
+
+
+class StudentsTPrior(_Prior):
+    density = "students"
+
+    def __init__(self, mean, standard_deviation, nu, lower=float("-inf"), upper=float("inf"), name=None):
+        super().__init__(name=name)
+        _validate_bounds(lower, mean, upper)
+        _validate_standard_deviation(standard_deviation)
+        _validate_nu(nu)
+
+        self.lower = lower
+        self.upper = upper
+        self.mean = mean
+        self.standard_deviation = standard_deviation
+        self.nu = nu
+
+    def _parameters(self):
+        return {
+            "lower": self.lower,
+            "upper": self.upper,
+            "mean": self.mean,
+            "std": self.standard_deviation,
+            "nu": self.nu,
+        }
+
+
+class LogGaussianPrior(_Prior):
+    density = "log_gaussian"
+
+    def __init__(self, mean, standard_deviation, eta, lower=float("-inf"), upper=float("inf"), name=None):
+        super().__init__(name=name)
+        _validate_bounds(lower, mean, upper)
+        _validate_standard_deviation(standard_deviation)
```

Comment:
I like seeing all these "validates".  It helps me; i.e. because it's not exactly textbook, i.e. not taught.

Reply:
Validation at API boundaries (which this is) is always a good idea.

Reply:
This is also nicer than the validations in Alex's code b/c they are validated in logically-related places, as opposed to a long list. Less brittle this way.